Buid break in CH Simulator

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### Informations

- **Qiskit Aer version**: 72863e804ec3d07c0da2ebeb665a31db7a7e3010
- **Python version**: 
- **Operating system**: macOS

### What is the current behavior?

In some build environment, `json_t` is not converted to `std::string`.
https://github.com/Qiskit/qiskit-aer/blob/72863e804ec3d07c0da2ebeb665a31db7a7e3010/src/simulators/ch/ch_runner.hpp#L615

```
qiskit-aer/src/third-party/headers/nlohmann_json.hpp:1201:9: error: static_assert failed "could not find from_json() method in T's namespace" [clang-diagnostic-error]
        static_assert(sizeof(BasicJsonType) == 0,
        ^
qiskit-aer/src/third-party/headers/nlohmann_json.hpp:1216:16: note: in instantiation of function template specialization 'nlohmann::detail::from_json_fn::call<nlohmann::basic_json<std::map, std::vector, std::__1::basic_string<char>, bool, long long, unsigned long long, double, std::allocator, adl_serializer>, std::__1::basic_string_view<char, std::__1::char_traits<char> > >' requested here
        return call(j, val, priority_tag<1> {});
               ^
qiskit-aer/src/third-party/headers/nlohmann_json.hpp:9669:9: note: in instantiation of function template specialization 'nlohmann::detail::from_json_fn::operator()<nlohmann::basic_json<std::map, std::vector, std::__1::basic_string<char>, bool, long long, unsigned long long, double, std::allocator, adl_serializer>, std::__1::basic_string_view<char, std::__1::char_traits<char> > >' requested here
        ::nlohmann::from_json(std::forward<BasicJsonType>(j), val);
        ^
qiskit-aer/src/third-party/headers/nlohmann_json.hpp:12094:36: note: in instantiation of function template specialization 'nlohmann::adl_serializer<std::__1::basic_string_view<char, std::__1::char_traits<char> >, void>::from_json<const nlohmann::basic_json<std::map, std::vector, std::__1::basic_string<char>, bool, long long, unsigned long long, double, std::allocator, adl_serializer> &, std::__1::basic_string_view<char, std::__1::char_traits<char> > >' requested here
        JSONSerializer<ValueType>::from_json(*this, ret);
                                   ^
qiskit-aer/src/third-party/headers/nlohmann_json.hpp:12355:16: note: in instantiation of function template specialization 'nlohmann::basic_json<std::map, std::vector, std::__1::basic_string<char>, bool, long long, unsigned long long, double, std::allocator, adl_serializer>::get<std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> >, 0>' requested here
        return get<ValueType>();
               ^
/usr/local/Cellar/llvm/7.0.0/bin/../include/c++/v1/string:875:29: note: in instantiation of function template specialization 'nlohmann::basic_json<std::map, std::vector, std::__1::basic_string<char>, bool, long long, unsigned long long, double, std::allocator, adl_serializer>::operator basic_string_view<std::__1::basic_string_view<char, std::__1::char_traits<char> >, 0>' requested here
        {__self_view __sv = __t; return assign(__sv);}
                            ^
qiskit-aer/src/simulators/ch/ch_runner.hpp:615:26: note: in instantiation of function template specialization 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::operator=<nlohmann::basic_json<std::map, std::vector, std::__1::basic_string<char>, bool, long long, unsigned long long, double, std::allocator, adl_serializer>, void>' requested here
    serialized_states[i] = serialize_state(i);
                         ^
```

### Steps to reproduce the problem

Build standalone.

```
mkdir build
cd build
cmake -DCMAKE_CXX_COMPILER=g++-8 ..
make
```

### What is the expected behavior?

Build finishes without errors.

### Suggested solutions

```
serialized_states[i] = serialize_state(i).dump();
```

--- a/src/simulators/ch/ch_runner.hpp
+++ b/src/simulators/ch/ch_runner.hpp
@@ -612,7 +612,7 @@ std::vector<std::string> Runner::serialize_decomposition() const
   #pragma omp parallel for if(num_threads_ > 1 && num_states_ > omp_threshold_) num_threads(num_threads_)
   for(uint_t i=0; i<num_states_; i++)
   {
-    serialized_states[i] = serialize_state(i);
+    serialized_states[i] = serialize_state(i).dump();
   }
   return serialized_states;
 }

